library/src/scala/CanEqual.scala

```diff
@@ -31,4 +31,13 @@ object CanEqual {
   // source code of these classes
   given canEqualSeq[T, U](using eq: CanEqual[T, U]): CanEqual[Seq[T], Seq[U]] = derived
   given canEqualSet[T, U](using eq: CanEqual[T, U]): CanEqual[Set[T], Set[U]] = derived
+
+  // The next three typeclass instances can also go into the companion objects of classes Option and Either.
+  // For now they are here in order not to have to touch the source code of these classes
+  given canEqualOptions[T, U](using eq: CanEqual[T, U]): CanEqual[Option[T], Option[U]] = derived
```

Comment:
I believe this can only be added in 3.1. Not sure if we can use `@experimnetal` here because the users will no have the choice to use it or not.

@sjrd @smarter WDYT?

Reply:
Indeed. This must be for 3.1.

Reply:
Thanks @nicolasstucki and @sjrd for your answers. I have a few questions. I'm so sorry to ask it when everyone is busy with finalizing Scala 3 for release and I really appreciate your answer.

1. Regarding the users having no choice once it's added, could you please think about who are affected by this?
   They are the ones who actually need `strictEquality` and most likely need this typeclass instance. The ones don't use `strictEquality` are not affected at all. Besides, the users do still have contol of `CanEqual[T, U]` since it's not auto-magically give them `CanEqual[T, U]` as well.
   
   Can there be another use case than this one? I think it would be hard to find any other use case of `CanEqual` for `Option` but I know that I could be easily wrong so I'm happy to take any advice. 

2. If what I mentioned above is not so important, that's fine. Could you tell me about the rest please?
   a. Can we have the other `CanEqual` for `Option` to solve `cannot be compared with == or !=` on `None` case in pattern matching?
   b. Can we have `CanEqual` for `Either`?
   c. Can we have `CanEqual` for `Tuple`?
 
**It will be so nice if at least the ones for `Tuple` are accepted because it's so common use case.**

Finally, could you please think about what would make `strictEquality` usable and more useful. Whenever users use `Option`, `Either` and `Tuple`, they need to import their own `CanEqual` typeclass instances seprately and it would be so annoying.

As far as I know, wildcard import like `import my.typeclasses._` doesn't work and it has to be done seprately like.
```scala
import my.typeclasses.{canEqualOptions, canEqualOption, canEqualEither, canEqualEmptyTuple, canEqualTuple}
```

Thank you!

Reply:
Oh, I just found that `3.0.0` has been tagged already. 😅

Reply:
Regarding,
> As far as I know, wildcard import like `import my.typeclasses._` doesn't work and it has to be done seprately like.

I didn't know that I could do `import my.typeclasses.given` for that. Sorry about lack of my Scala 3 knowledge. 😅